### ISSUE TITLE
Bug: SentryReporter cannot be Instanciated

### ISSUE THREAD

**Issue by 749** [esbenp/heimdal#8]
PHP 5.6.24 (cli) (built: Jul 20 2016 21:19:29)
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies

Laravel version v5.4.15

`[2017-03-17 07:48:09] local.ERROR: exception 'Illuminate\Contracts\Container\BindingResolutionException' with message 'Unresolvable dependency resolving [Parameter #0 [ <required> array $config ]] in class Optimus\Heimdal\Reporters\SentryReporter' in C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Container\Container.php:819
Stack trace:
#0 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Container\Container.php(757): Illuminate\Container\Container->unresolvablePrimitive(Object(ReflectionParameter))
#1 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Container\Container.php(732): Illuminate\Container\Container->resolvePrimitive(Object(ReflectionParameter))
#2 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Container\Container.php(710): Illuminate\Container\Container->resolveDependencies(Array)
#3 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Container\Container.php(565): Illuminate\Container\Container->build('Optimus\\Heimdal...')
#4 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(702): Illuminate\Container\Container->make('Optimus\\Heimdal...')
#5 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\optimus\heimdal\src\ExceptionHandler.php(74): Illuminate\Foundation\Application->make('Optimus\\Heimdal...', Array)
#6 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(312): Optimus\Heimdal\ExceptionHandler->report(Object(Illuminate\Contracts\Container\BindingResolutionException))
#7 C:\Users\Jan\Documents\Projects\laravel\cms_manager\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(118): Illuminate\Foundation\Http\Kernel->reportException(Object(Illuminate\Contracts\Container\BindingResolutionException))
#8 C:\Users\Jan\Documents\Projects\laravel\cms_manager\public\index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#9 C:\Users\Jan\Documents\Projects\laravel\cms_manager\server.php(21): require_once('C:\\Users\\Jan\\Do...')
#10 {main}  `

**Comment by esbenp** [comment]
Can you pass your heimdal config? Of course remember to remove api secrets :-)

**Comment by 749** [comment]
I did not change the api just uncommented the lines for sentry and added the keys

**Comment by esbenp** [comment]
I am sorry, it is going to be really hard for me to debug this if you do not pass your configuration here :-)

**Comment by 749** [comment]
Sorry I meant to say that I did not change the config other than uncommenting the sentry configuration part. here's what I did to install:

```
λ composer require optimus/heimdal
Using version ^1.2 for optimus/heimdal
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing optimus/heimdal (1.2.2) Downloading: 100%
optimus/heimdal suggests installing asm89/stack-cors (Needed to add CORS headers if add_cors_headers configuration is true)
optimus/heimdal suggests installing sentry/sentry (Needed to report exceptions to Sentry.io)
Writing lock file
Generating autoload files
> Illuminate\Foundation\ComposerScripts::postUpdate
> php artisan ide-helper:generate
A new helper file was written to _ide_helper.php
> php artisan ide-helper:meta
A new meta file was written to .phpstorm.meta.php
> php artisan optimize
Generating optimized class loader
The compiled services file has been removed.


λ composer require sentry/sentry
Using version ^1.6 for sentry/sentry
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing sentry/sentry (1.6.2) Loading from cache
Writing lock file
Generating autoload files
> Illuminate\Foundation\ComposerScripts::postUpdate
> php artisan ide-helper:generate
A new helper file was written to _ide_helper.php
> php artisan ide-helper:meta
A new meta file was written to .phpstorm.meta.php
> php artisan optimize
Generating optimized class loader
The compiled services file has been removed.

// Added heimdal to config/app.php
// replaced in bootstrap/app.php
$app->singleton(
    Illuminate\Contracts\Debug\ExceptionHandler::class,
    Brace\Exceptions\Handler::class
);
// with
$app->singleton(
    Illuminate\Contracts\Debug\ExceptionHandler::class,
    Optimus\Heimdal\ExceptionHandler::class
);

λ php artisan vendor:publish --provider="Optimus\Heimdal\Provider\LaravelServiceProvider"
Copied File [\vendor\optimus\heimdal\src\config\optimus.heimdal.php] To [\config\optimus.heimdal.php]
Publishing complete.

//uncommented config for sentry in heimdal and stored dsn in .env file

//added throw exception to main page:
throw new BackupException('just a test');

// and i get the error described above

```

here's the config I used:
```php
<?php

use Symfony\Component\HttpKernel\Exception as SymfonyException;
use Optimus\Heimdal\Formatters;

return [
    'add_cors_headers' => false,

    // Has to be in prioritized order, e.g. highest priority first.
    'formatters' => [
        SymfonyException\UnprocessableEntityHttpException::class => Formatters\UnprocessableEntityHttpExceptionFormatter::class,
        SymfonyException\HttpException::class => Formatters\HttpExceptionFormatter::class,
        Exception::class => Formatters\ExceptionFormatter::class,
    ],

    'response_factory' => \Optimus\Heimdal\ResponseFactory::class,

    'reporters' => [
        'sentry' => [
            'class'  => \Optimus\Heimdal\Reporters\SentryReporter::class,
            'config' => [
                'dsn' => env('SENTRY_DSN'),
                // For extra options see https://docs.sentry.io/clients/php/config/
                // php version and environment are automatically added.
                'sentry_options' => []
            ]
        ]
    ],

    'server_error_production' => 'An error occurred.'
];

```

**Comment by 749** [comment]
I think Laravel has an issue when trying to push the configuration into the SentryReporter class, maybe the config is not of type array?

**Comment by ghost** [comment]
Any update on this? I've the same issue.

**Comment by ghost** [comment]
@749 The config is correctly passed as an array but the Container won't resolve it.

**Comment by esbenp** [comment]
Did you remember to install the Sentry library using `composer require sentry/sentry`?

**Comment by ghost** [comment]
Of course I did.

**Comment by esbenp** [comment]
Alright it seems Laravel removed the option passing parameters when resolving dependencies using the container in 5.4. See a discussion here https://github.com/laravel/internals/issues/391 
I fixed this in #12 by moving reporter resolving to the service provider.

The fix has been deployed in `1.3`

@aniljaiswal @749 care to give it a try?
